lib/puppet/provider/agent_upgrade/windows.rb

```diff
@@ -0,0 +1,85 @@
+
+Puppet::Type.type(:agent_upgrade).provide(:agent_upgrade) do
+  confine :osfamily => /windows/i
+
+  mk_resource_methods
```

Comment:
This creates the accessor methods for properties allowing us to set them for self.instances